compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1216,6 +1216,20 @@ impl<'a> Parser<'a> {
 
     /// Parses an enum declaration.
     fn parse_item_enum(&mut self) -> PResult<'a, ItemInfo> {
+        if self.prev_token.is_keyword(kw::Enum) && self.token.is_keyword(kw::Struct) {
+            let mut err = self.struct_span_err(
+                self.prev_token.span.to(self.token.span),
+                "`enum` and `struct` are mutually exclusive",
+            );
+            err.span_suggestion(
+                self.prev_token.span.to(self.token.span),
+                "replace `enum struct` with `enum`",
+                "enum",
+                Applicability::MachineApplicable,
+            );
+            err.emit();
```

Comment:
You need to check if the next token is an identifier. 

You can use the lookahead method on the parser to do this. If it is an identifier, then you need to eat the current token, _then_ call `.emit()`.

If not, then you need to just `return Err(err);`